chat/src/main/java/com/github/twitch4j/chat/TwitchChatBuilder.java

```diff
@@ -284,16 +296,17 @@ public TwitchChat build() {
         if (credentialManager == null) {
             credentialManager = CredentialManagerBuilder.builder().build();
         }
+        TwitchAuth.registerIdentityProvider(credentialManager, clientId, clientSecret, null);
 
         // Register rate limits across the user id contained within the chat token
         final String userId;
         if (chatAccount == null) {
             userId = null;
         } else {
             if (StringUtils.isEmpty(chatAccount.getUserId())) {
-                chatAccount = credentialManager.getOAuth2IdentityProviderByName("twitch")
-                    .orElse(new TwitchIdentityProvider(null, null, null))
-                    .getAdditionalCredentialInformation(chatAccount).orElse(chatAccount);
+                credentialManager.getIdentityProviderByName("twitch", TwitchIdentityProvider.class)
```

Comment:
if we go back to using `orElse` here, we don't *need* `TwitchAuth` to throw exception on dupe provider name w/ different class